### PR TITLE
feat: Date and timestamp trunc with format array

### LINKED ISSUE #93: date and timestamp trunc should support an array of formats
### Describe the bug

Spark's `trunc` and `date_trunc` functions take a `format`  as input where the `format` may be either a scalar string or an array of strings.
Comet's current implementation only supports a `format` that is a scalar string. 

### Steps to reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by parthchandra**
feat: date and timestamp trunc with format array

## Which issue does this PR close?

Closes #93 

## Rationale for this change
Compatibility with Spark

## What changes are included in this PR?
Implements support for the `format` parameter to be an array.

## How are these changes tested?

Adds additional unit tests

### CODE REVIEW

**Review by viirya — APPROVED** [review]
I think this has been reviewed internally.

**Comment by parthchandra** [comment]
@viirya thank you for the approval (and review). Will wait for other comments and ci before merging.

**Review by sunchao — APPROVED** [review]
LGTM (pending CI)

**Comment by sunchao** [comment]
Merge, thanks

**Pull request merged**
